### PR TITLE
Support cpu hotplug adding/removing on arm64 kata-containers

### PR DESCRIPTION
**Pull request by justin-he**

### CODE REVIEW

**Comment by justin-he** [comment]
Already tested by:
docker run --runtime=kata-runtime -ti --cpus 1 debian bash
get the container id by docker ps 
 docker update --cpus 2 $container_id
 docker update --cpus 1 $container_id

**Comment by justin-he** [comment]
There will be 3 patch parts of this feature:
1. qemu side
2. runtime side
3. agent side

**Comment by jodh-intel** [comment]
Marking as dnm as blocked on https://github.com/kata-containers/agent/pull/478

File: virtcontainers/sandbox.go [hunk 1]
```diff
@@ -1638,6 +1639,17 @@ func (s *Sandbox) updateResources() error {
 			return err
 		}
 	}
+
+	// on arm64, need to handle the offline case
+	if runtime.GOARCH == "arm64" {
```
[the remark this answers is not in the document]
Comment (justin-he) [reply virtcontainers/sandbox.go]: Hi, @jodh-intel for this point, do you think x86 and ppc64le need the offline feature?
If yes, I would remove that line "if runtime.GOARCH == "arm64" {" , but kernel the offline process in sandbox.go

File: pkg/katautils/config.go [hunk 2]
```diff
@@ -257,7 +257,13 @@ func (h hypervisor) defaultMaxVCPUs() uint32 {
 	//don't exceed the number of physical CPUs. If a default is not provided, use the
 	// numbers of physical CPUs
 	if reqVCPUs >= numcpus || reqVCPUs == 0 {
-		reqVCPUs = numcpus
+		if goruntime.GOARCH != "arm64" {
```
[the remark this answers is not in the document]
Comment (justin-he) [reply pkg/katautils/config.go]: ok

File: virtcontainers/sandbox.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jodh-intel) [reply virtcontainers/sandbox.go]: If the architecture supports it, I would say yes as we want to keep the functionality consistent across architectures where possible.

/cc @devimc, @nitkon, @alicefr.

**Comment by raravena80** [comment]
@justin-he Hi, any updates? Thx!

**Comment by justin-he** [comment]
Hi @raravena80 thanks for the reminding， I have finished the patchv2 for kata runtime an am waiting for the agent pr result.
Shouldn't I wait for the kata-agent to merge my pr firstly? [agent pr](https://github.com/kata-containers/agent/pull/478)
Sorry if this is a dumb question :)

**Comment by sboeuf** [comment]
@justin-he yes the agent PR would have to go first. And on the agent PR, we have some ongoing discussion to determine if this is actually acceptable design for Kata.

**Comment by raravena80** [comment]
@justin-he nudge

**Comment by justin-he** [comment]
close it temporarily since this runtime side pr is dependent on agent side which is still need approval

**Pull request closed without merging**